### PR TITLE
Increase mem for merge_gvcfs_job and make storage_gb configurable for _haplotype_caller_one

### PR DESCRIPTION
**Pull request by cassimons**
Two small changes for genotype jobs:

* Increase mem for merge_gvcfs_jobs to prevent out-of-memory errors on output. As discussed [here](https://centrepopgen.slack.com/archives/C051EPKNJH1/p1686733537921009).
* Attached storage is a large cost driver for _haplotype_caller_one, the single most expensive step in the WGS pipeline. This drops the default to 40gb which should work for 99% of genomes. By making it configurable, it will be easy to set higher for any needed samples.

### CODE REVIEW

**Review by MattWellie — APPROVED** [review]

File: cpg_workflows/jobs/genotype.py [hunk 1]
```diff
@@ -189,6 +189,15 @@ def _haplotype_caller_one(
     # of a Hail worker (2 cores), plus with the `highmem` instance that would
     # give enough memory: 13G. That's not going to give enough disk storage, so we
     # are explicitly requesting more storage.
+    #
+    # Based on an audit of RD crams on 19/05/23, 99% of crams are <34Gb. Will set the
+    # default to 40Gb for genomes then use a run specific confg to run the rare
+    # sample that will fail from this limit.
```
Comment (vivbak) [inline cpg_workflows/jobs/genotype.py:195]:
```suggestion
    # sequencing group that will fail from this limit.
```
In anticipation for this change coming v. soon https://github.com/populationgenomics/production-pipelines/pull/366

**Review by vivbak — APPROVED** [review]

**Comment by jmarshall** [comment]
I just looked at this one again, and surely it's not going to be effective because `storage_gb` is still being set back to `None`/`100` immediately after the lines added?

**Comment by cassimons** [comment]
Hmmm... that is not so great! The memory fix worked at least.  I will put in an new PR.

**Comment by cassimons** [comment]
@vivbak from #383 and before I fix the dumb mistake pointed out by John above:
> can we add this new param to the defaults.toml file with a comment on what it's for.

`haplotypecaller_storage_gb` is only being applied for genomes not exomes. If I am going to move it to defaults.toml should I rename it to something like `haplotypecaller_storage_gb_genome`? Do I need to add an equivalent for exomes or is that ok to leave in the logic of the job?

**Comment by vivbak** [comment]
@cassimons Perhaps better to put it somewhere nested like [workflows.genome] in the config, to make the distinction clear. Then you can access it with := and it might simplify a little. That way it's clear its genome only without the var name becoming too verbose.

**Comment by MattWellie** [comment]
@cassimons I've bundled a candidate change for this into #383 
- if there's a manual override setting, use that
- if not and it's a genome, 40GB
- else, no extra storage

**Pull request merged**